dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt

```diff
@@ -0,0 +1,353 @@
+      ).toString()
+    val initialState = InitialState(
+      suffixData = createOp.suffixData,
+      delta = createOp.delta,
+    )
+    val longFormDidSegment = didUriSegment(initialState)
+
+    val response: HttpResponse = runBlocking {
+      client.post(operationsEndpoint) {
+        contentType(ContentType.Application.Json)
+        setBody(createOp)
+      }
+    }
+
+    val opBody = runBlocking {
+      response.bodyAsText()
+    }
+
+    if (response.status.isSuccess()) {
+      val shortFormDid = "did:ion:$shortFormDidSegment"
+      val longFormDid = "$shortFormDid:$longFormDidSegment"
+      val resolutionResult = resolve(longFormDid)
+
+      if (!resolutionResult.didResolutionMetadata?.error.isNullOrEmpty()) {
+        throw ResolutionException(
+          "error when resolving after creation: ${resolutionResult.didResolutionMetadata?.error}"
+        )
+      }
+
+      return DidIonHandle(
+        resolutionResult.didDocument.id.toString(),
+        keyManager,
+        IonCreationMetadata(
+          createOp,
+          shortFormDid,
+          longFormDid,
+          opBody,
+          keys
+        )
+      )
+    }
+    throw InvalidStatusException("received error response '$opBody'")
+  }
+
+  private inline fun <reified T> canonicalized(data: T): ByteArray {
+    val jsonString = mapper.writeValueAsString(data)
+    return JsonCanonicalizer(jsonString).encodedUTF8
+  }
+
+  private inline fun <reified T> didUriSegment(initialState: T): String {
+    val canonicalized = canonicalized(initialState)
+    return Base64URL.encode(canonicalized).toString()
+  }
+
+  /**
+   * Given a [did], returns the [DidResolutionResult], which is specified in https://w3c-ccg.github.io/did-resolution/#did-resolution-result
+   *
+   * @throws [InvalidStatusException] When any of the network requests return an invalid HTTP status code.
+   */
+  override fun resolve(did: String, options: ResolveDidOptions?): DidResolutionResult {
+    val didObj = DID.fromString(did)
+    require(didObj.methodName == methodName)
+
+    val resp = runBlocking { client.get("$identifiersEndpoint/$didObj") }
+    val body = runBlocking { resp.bodyAsText() }
+    if (!resp.status.isSuccess()) {
+      throw InvalidStatusException("resolution error response '$body'")
+    }
+    return mapper.readValue(body, DidResolutionResult::class.java)
+  }
+
+  private fun createOperation(keyManager: KeyManager, options: CreateDidIonOptions?)
+    : Pair<SidetreeCreateOperation, KeyAliases> {
+    val updatePublicJWK: JWK = if (options?.updatePublicJWK == null) {
+      val alias = keyManager.generatePrivateKey(JWSAlgorithm.ES256K, Curve.SECP256K1)
+      keyManager.getPublicKey(alias)
+    } else {
+      options.updatePublicJWK
+    }
+    val publicKeyCommitment: String = publicKeyCommitment(updatePublicJWK)
+
+    val verificationMethodId = options?.verificationMethodId ?: UUID.randomUUID().toString()
+    validateVerificationMethodId(verificationMethodId)
```

Comment:
```suggestion
    val verificationMethodId = when (options?.verificationMethodId) {
      null -> UUID.randomUUID().toString()
      else -> {
        validateVerificationMethodId(options.verificationMethodId)
        options.verificationMethodId
      }
    }
```

if you want to prevent validating something you just created